integration/integration_test.go

```diff
@@ -309,6 +312,89 @@ func TestBuildViaRegistryMirror(t *testing.T) {
 	checkContainerDiffOutput(t, diff, expected)
 }
 
+func TestBuildWithLocalTars(t *testing.T) {
```

Comment:
@JordanGoasdoue This is really great!! 
Thank you for submitting this PR!

Can you move this from integration test to a unit test and provide tests 
1. valid tar.gz
2. invalid tar.gz
3. tar.gz does not exists

Reply:
Hey @tejal29 , thank you very much for your comment and review, it's a pleasure to help on such project :) !

I tried to do the Unit testing instead of the integration test, but I have some troubles about it...

Here is my temporary directory after i've run the `make test` command : 
```
/tmp/013179842> ls -h
Dockerfile_invalid  Dockerfile_invalid.tar.gz  Dockerfile_non_existing  Dockerfile_valid  Dockerfile_valid.tar.gz
```

It returns me this, it seems to not even go on my function `TestBuildWithLocalTar` : 
```
coverage: 52.0% of statements
ok      github.com/GoogleContainerTools/kaniko/pkg/util (cached)        coverage: 52.0% of statements
?       github.com/GoogleContainerTools/kaniko/pkg/version      [no test files]
?       github.com/GoogleContainerTools/kaniko/testutil [no test files]
FAIL
Makefile:62: recipe for target 'test' failed
make: *** [test] Error 1
```

When I run myself kaniko in docker to simulate the result, this is what I have : 
Dockerfile_valid : 
```
docker run --net=host -v /tmp/013179842:/workspace -v /home/jordan/.config/gcloud:/root/.config/gcloud executor-image -f Dockerfile_valid -d localhost:5000/kaniko-test-local --no-push -c tar://Dockerfile_valid.tar.gz
INFO[0000] Resolved base name debian:9.11 to debian:9.11 
INFO[0000] Resolved base name debian:9.11 to debian:9.11
```
Dockerfile_invalid : 
```
docker run --net=host -v /tmp/013179842:/workspace -v /home/jordan/.config/gcloud:/root/.config/gcloud executor-image -f Dockerfile_invalid -d localhost:5000/kaniko-test-local --no-push -c tar://Dockerfile_invalid.tar.gz
Error: error resolving source context: EOF
```
Dockerfile_non_existing : 
```
docker run --net=host -v /tmp/013179842:/workspace -v /home/jordan/.config/gcloud:/root/.config/gcloud executor-image -f Dockerfile_non_existing -d localhost:5000/kaniko-test-local --no-push -c tar://Dockerfile_non_existing.tar.gz
Error: error resolving source context: open Dockerfile_non_existing.tar.gz: no such file or directory
```

What am I doing wrong here ?

My code is updated if you want to look at it

Thanks

Reply:
Ok finally I can see the error with Travis : 

https://travis-ci.org/github/GoogleContainerTools/kaniko/jobs/661273579?utm_medium=notification&utm_source=github_status : 

`tar_test.go:106: Unexpected error: unpacking tar from build context: mkdir /kaniko: permission denied`

Any proposition here to do this test without having this kind of error ?